### ISSUE TITLE
Not compatible RN 0.43?

### ISSUE THREAD

**Issue by XavierColombel** [archriss/react-native-snap-carousel#54]
I've got these errors:

```
npm WARN react-addons-shallow-compare@15.5.0-alpha.0 requires a peer of react@^15.5.0 but none was installed.
npm WARN react-test-renderer@15.4.2 requires a peer of react@^15.4.2 but none was installed.
```

![capture d ecran 2017-04-04 a 15 27 58](https://cloud.githubusercontent.com/assets/1059964/24658926/802379c2-194b-11e7-908d-3ba46bf7a5b2.png)

**Comment by bd-arc** [comment]
@XavierColombel That's because RN 0.43 uses React version `~16.0.0-alpha.3`, while our plugin dependencies expect version `15.x.y`. This latest version of React [hasn't been documented yet](https://github.com/facebook/react/releases), so I don't know if we should expect breaking changes.

Note that we haven't tested compatibility with RN 0.43 since it hasn't been officially released yet. My guess is that it should work flawlessly though. You should just ignore npm warnings, try the plugin, and get back to me if it doesn't work.

**Comment by XavierColombel** [comment]
Ok I see what you mean. I've created a brand new project and I've added react-native-snap-carousel only, in order to test it. It doesn't work with RN 0.43 (see the red screen).

**Comment by bd-arc** [comment]
@XavierColombel Thanks for the link (you should edit it by the way, since it redirects to a 404 error page ;-). Here is what seems to be the plan for the `shallow-compare` addon:

> Plan for 16: we won’t be releasing 16 for this package. Nevertheless version 15 should keep working indefinitely.

Still, I don't like that part:

> Internal dependencies (**will break in 16**)

The issue is that `react-addons-shallow-compare` wasn't installed with our plugin, which probably has to do with those changes. We'll try to migrate to RN 0.43 and see what we can do about it.

**Comment by XavierColombel** [comment]
@bd-arc I just updated the link! ;) Ok, so I wait for a RN 0.43 compatible version. Thanks for your help.

**Comment by gaearon** [comment]
Which version of `react` package are you using?

**Comment by XavierColombel** [comment]
With RN 0.43, by default:

```
"dependencies": {
    "react": "16.0.0-alpha.6",
    "react-native": "0.43.0"
}
```

**Comment by gaearon** [comment]
And which versions of addons?

**Comment by gaearon** [comment]
(Can you double check their `package.json` in `node_modules` please)

**Comment by bvaughn** [comment]
Chiming in kind of late, but here's a bit more context:

As you've noticed, `shallowCompare` is no longer accessible from `react/lib/shallowCompare` for react v16+. Going forward the shallow-compare code has been moved _into_ its own package (instead of just forwarding to react). This change was made in `react-addons-shallow-compare@15.5.0-alpha.0`.

We wanted to update the version of react used inside of react-native and so we needed to also update to a newer release of `react-addons-shallow-compare`. Unfortunately there was a `peerDependencies` in the alpha release I pinned to, which is why you're seeing that warning. (That has been corrected in the subsequent `react-addons-shallow-compare@15.5.0` release.)

That being said, you shouldn't be seeing the reference to `react/lib/shallowCompare` from `react-addons-shallow-compare` in the error screenshot you attached- unless one of your plugins is not using `react-addons-shallow-compare@15.5.0-alpha.0`.

**Comment by bd-arc** [comment]
@bvaughn Thank you very much for the detailed information! I appreciate that. I'll make the necessary adjustments to our plugin.

Since `shallowCompare` [has apparently been deprecated](https://facebook.github.io/react/docs/shallow-compare.html), would you recommend migrating to `React.PureComponent` instead?

**Comment by bvaughn** [comment]
No problem! 😄 

`react-addons-shallow-compare` should continue to work indefinitely, even though we don't plan any future releases of the add-on past version 15.5.0.

`PureComponent` is the recommended path going forward though. If you can easily switch, go for it!

**Comment by bd-arc** [comment]
@XavierColombel Issue should be resolved with version 2.1.1. Can you try and confirm?

@gaearon @bvaughn Thanks for your help with this matter ;)

**Comment by XavierColombel** [comment]
@bd-arc You rock! Thank you so much!
